src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollectionOfT.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Represents an <see cref="IEdmObject"/> that is a collection of <see cref="IEdmChangedObject"/>s.
+    /// </summary>
+    [NonValidatingParameterBinding]
+    public class EdmChangedObjectCollection<TStructuralType> : EdmChangedObjectCollection, ICollection<IEdmChangedObject<TStructuralType>>, IEdmObject
+    {
+        private IEdmEntityType _entityType;
+        private EdmDeltaCollectionType _edmType;
+        private IEdmCollectionTypeReference _edmTypeReference;
+        ICollection<IEdmChangedObject<TStructuralType>> _items;
```

Comment:
_entityType, _edmType, and _edmTypeReference are set but don't appear to be used anywhere  in the derived type. Do we need them here?